### PR TITLE
Add salvage item fuctionality into equipment popover

### PR DESCRIPTION
**Pull request by ghost**
Adds the fuctionality requested by #134

### CODE REVIEW

File: src/client/app/inventory/item.popover.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class InventoryItemPopover {
   }
 
   salvage() {
-    this.socketService.emit(ServerEventName.ItemSalvage, { itemId: this.item.id });
+    this.socketService.emit(ServerEventName.ItemSalvage, { itemId: this.item.id, ItemSlot: null });
```
Comment (seiyria) [inline src/client/app/inventory/item.popover.ts]: this should probably pass in `itemSlot`, not `ItemSlot`

File: src/server/modules/inventory.ts [hunk 2]
```diff
@@ -72,19 +72,31 @@ export class SellItemEvent extends ServerSocketEvent implements ServerEvent {
...
+  args = 'itemId?, itemSlot?';
 
-  async callback({ itemId } = { itemId: '' }) {
+  async callback({ itemId, itemSlot } = { itemId: '', itemSlot: '' }) {
     const player = this.player;
     if(!player) return this.notConnected();
 
-    const foundItem = player.$inventory.getItemFromInventory(itemId);
-    if(!foundItem) return this.gameError('Could not find that item in your inventory.');
+    const foundItem = (function() {
```
Comment (seiyria) [inline src/server/modules/inventory.ts]: please don't use anonymous functions like this. just use a normal if statement, as well as `let`.

File: src/server/modules/inventory.ts [hunk 3]
```diff
@@ -72,19 +72,31 @@ export class SellItemEvent extends ServerSocketEvent implements ServerEvent {
...
+      if(itemSlot == null) {
+        return player.$inventory.getItemFromInventory(itemId);
+      } else {
+        return player.$inventory.itemInEquipmentSlot(<ItemSlot>itemSlot);
+      }
+    })()
+
+    if(!foundItem) return this.gameError('Could not find that item in your inventory or equiped gear.');
```
Comment (seiyria) [inline src/server/modules/inventory.ts]: typo: `equipped`

File: src/server/modules/inventory.ts [hunk 4]
```diff
@@ -72,19 +72,31 @@ export class SellItemEvent extends ServerSocketEvent implements ServerEvent {
...
 
     if(foundItem.locked) return this.gameError('Item is currently locked. Unlock it to salvage it.');
 
     const { wood, stone, clay, astralium } = player.salvageItem(foundItem);
-    player.$inventory.removeItemFromInventory(foundItem);
+
+    if(itemSlot == null) {
```
Comment (seiyria) [inline src/server/modules/inventory.ts]: don't use `==`. the linter should catch this. you can just use `!itemSlot`

File: src/server/modules/inventory.ts [hunk 5]
```diff
@@ -72,19 +72,31 @@ export class SellItemEvent extends ServerSocketEvent implements ServerEvent {
...
 
-  async callback({ itemId } = { itemId: '' }) {
+  async callback({ itemId, itemSlot } = { itemId: '', itemSlot: '' }) {
     const player = this.player;
     if(!player) return this.notConnected();
 
-    const foundItem = player.$inventory.getItemFromInventory(itemId);
-    if(!foundItem) return this.gameError('Could not find that item in your inventory.');
+    const foundItem = (function() {
+      if(itemSlot == null) {
```
Comment (seiyria) [inline src/server/modules/inventory.ts]: don't check against `null` like this. similar to below.

**Comment by seiyria** [comment]
Looks good, I have some feedback and it needs to pass the travis CI tests. Thanks for your work so far!

File: src/server/modules/inventory.ts (hunk 3 above)
Comment (ghost) [reply src/server/modules/inventory.ts]: typo fixed

File: src/server/modules/inventory.ts (hunk 2 above)
Comment (ghost) [reply src/server/modules/inventory.ts]: I wanted to keep it as `const`, but I see how that was counter-productive.

File: src/client/app/inventory/item.popover.ts (hunk 1 above)
Comment (ghost) [reply src/client/app/inventory/item.popover.ts]: Indeed, a small typo on my part. It's fixed now.

File: src/server/modules/inventory.ts (hunk 4 above)
Comment (ghost) [reply src/server/modules/inventory.ts]: Yep! Fixed.

File: src/server/modules/inventory.ts (hunk 5 above)
Comment (ghost) [reply src/server/modules/inventory.ts]: Same as the other null check. Fixed.

**Pull request merged**